features/steps/exporter_s3.py

```diff
@@ -12,94 +12,123 @@
-    context.minio_secret_access_key = value
+    context.S3_secret_access_key = value
 
 
-@given(u"Minio bucket name is set to {value}")
-def set_minio_bucket_name(context, value):
-    """Set Minio bucket name."""
+@given("S3 bucket name is set to {value}")
+def assert_s3_bucket_name_is_set(context, value):
+    """Set S3 bucket name."""
     assert value is not None, "Bucket name needs to be specified"
-    context.minio_bucket_name = value
+    context.S3_bucket_name = value
+
 
+@given("S3 connection is established")
+def establish_s3_connection(context):
+    """Establish connection to S3."""
+    minio_client(context)
 
-@then(u"I should see following objects generated in S3")
+
+@then("I should see following objects generated in S3")
 def check_objects_in_s3(context):
     """Check that all specified objects was generated."""
-    # construct new Minio client
-    client = minio_client(context)
-
     # check if bucket used by exporter exists
-    bucket_check(context, client)
+    bucket_check(context)
 
     # retrieve all objects stored in bucket
-    objects = client.list_objects(context.minio_bucket_name, recursive=False)
-
+    objects = context.minio_client.list_objects(context.S3_bucket_name, recursive=True)
     # retrieve object names only
     names = [o.object_name for o in objects]
+    print("S3 objects: ", names)
 
     # iterate over all items in feature table
     for row in context.table:
-        object_name = row["File name"]
-        assert object_name in names, \
-            "Can not find object {} in bucket {}".format(object_name, context.minio_bucket_name)
+        object_name = f"{context.S3_bucket_name}/{row['File name']}"
+        assert object_name in names, "Can not find object {} in bucket {}".format(
+            object_name, context.S3_bucket_name
+        )
 
 
-@then(u"I should see following number of records stored in CSV objects in S3")
+@then("I should see following number of records stored in CSV objects in S3")
 def check_csv_content_in_s3(context):
     """Check content of objects stored in S3."""
-    # construct new Minio client
-    client = minio_client(context)
-
     # check if bucket used by exporter exists
-    bucket_check(context, client)
+    bucket_check(context)
 
     # iterate over all items in feature table
     for row in context.table:
-        object_name = row["File name"]
+        object_name = f"{context.S3_bucket_name}/{row['File name']}"
         expected_records = int(row["Records"])
 
         # read object content
-        buff = read_object_into_buffer(context, client, object_name)
+        buff = read_object_into_buffer(context, object_name)
 
         # read CVS from buffer
-        csvFile = csv.reader(buff)
+        csv_file = csv.reader(buff)
 
         # skip the first row of the CSV file.
-        next(csvFile)
+        next(csv_file)
 
         stored_records = 0
-        for lines in csvFile:
+        for _ in csv_file:
             stored_records += 1
```

Comment:
Nitpick: maybe `len(csv_file) - 1` is slightly better...

Reply:
actually it is. I didn't stop to think about it, just removed the variable as it was unused =)